crates/amaru-ledger/src/rules/block.rs

```diff
@@ -47,42 +50,69 @@ pub enum InvalidBlockDetails {
         transaction_index: u32,
         violation: InvalidTransaction,
     },
-    UncategorizedError(String),
 }
 
 #[derive(Debug)]
-pub enum BlockValidation {
-    Valid,
+pub enum BlockValidation<A, E> {
+    Valid(A),
```

Comment:
Is `A` used at all? Or that's for some future expected behavior?

Reply:
It's used in the header size rule!